server/text_generation_server/layers/fp8.py

```diff
@@ -39,26 +55,70 @@ def fp8_quantize(weight, qdtype=torch.float8_e4m3fn):
 
 
 @dataclass
-class Fp8Weight(Weight):
+class Fp8Weight:
     weight: torch.Tensor
+    dtype: torch.dtype
+    weight_scale: Optional[torch.Tensor] = None
+    input_scale: Optional[torch.Tensor] = None
 
     def get_linear(self, bias: torch.Tensor):
-        return get_fp8_linear()(self.weight, bias)
+        if self.weight_scale is None:
+            return get_fp8_linear().from_unquant(self.weight, bias, self.dtype)
+        return get_fp8_linear().from_fp8(
+            self.weight, self.weight_scale, self.input_scale, bias, self.dtype
+        )
```

Comment:
Nice! Looks like a pattern we could reuse in the future for quantization that is pre-quantized or on the fly.